managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -141,7 +141,11 @@ public void safeRun() {
         ManagedLedgerImpl.TOTAL_SIZE_UPDATER.addAndGet(ml, dataLength);
         if (ml.hasActiveCursors()) {
             // Avoid caching entries if no cursor has been created
-            ml.entryCache.insert(new EntryImpl(ledger.getId(), entryId, data));
+            EntryImpl entry = EntryImpl.create(ledger.getId(), entryId, data);
+            // EntryCache.insert: duplicates entry by allocating new entry and data. so, recycle entry after calling
+            // insert
+            ml.entryCache.insert(entry);
+            entry.recycle();
```

Comment:
As we discussed, `ml.entryCache.insert(entry)` creates new `EntryImpl` internally and allocates new data to store into cache. So, life-cycle of this entry is finished after calling `cache.insert(entry)` so, recycling of this entry will not affect cache data because cache is already having duplicate-created entry.